File: Util/ManejoErrores.py
import logging
import traceback
import os
from PyQt6.QtWidgets import QMessageBox

class ManejoErrores:
    RUTA_LOG = "corporativo_pos_errores.log"
    LOGGER = logging.getLogger("CorporativoPOS")
    LOGGER.setLevel(logging.DEBUG)
    LOGGER.propagate = False
    
    _manejador_configurado = False
    
    if not _manejador_configurado:
        try:
            handler = logging.FileHandler(RUTA_LOG, mode='a')
            formatter = logging.Formatter('[%(asctime)s] [%(levelname)s] %(message)s', datefmt='%d/%m/%Y %H:%M:%S')
            handler.setFormatter(formatter)
            LOGGER.addHandler(handler)
            _manejador_configurado = True
        except Exception as e:
            print("No se pudo inicializar el archivo de log:", e)

    @staticmethod
    def error(padre, titulo, mensaje, ex=None):
        if ex is None:
            ManejoErrores.registrar(logging.ERROR, f"{titulo}: {mensaje}")
            QMessageBox.critical(padre, titulo, mensaje)
        else:
            ManejoErrores.registrarError(f"{titulo}: {mensaje}", ex)
            QMessageBox.critical(padre, titulo, f"{mensaje}\n\nDetalle: {str(ex)}")

    @staticmethod
    def advertencia(padre, titulo, mensaje):
        ManejoErrores.registrar(logging.WARNING, f"{titulo}: {mensaje}")
        QMessageBox.warning(padre, titulo, mensaje)

    @staticmethod
    def info(padre, titulo, mensaje):
        QMessageBox.information(padre, titulo, mensaje)

    @staticmethod
    def confirmar(padre, titulo, mensaje):
        resp = QMessageBox.question(padre, titulo, mensaje, QMessageBox.StandardButton.Yes | QMessageBox.StandardButton.No)
        return resp == QMessageBox.StandardButton.Yes

    @staticmethod
    def registrar(nivel, mensaje):
        ManejoErrores.LOGGER.log(nivel, mensaje)

    @staticmethod
    def registrarError(contexto, ex):
        tb = traceback.format_exc()
        ManejoErrores.LOGGER.error(f"{contexto}\n{tb}")

    @staticmethod
    def registrarInfo(evento):
        ManejoErrores.LOGGER.info(evento)

    @staticmethod
    def validarRequerido(padre, valor, nombreCampo):
        if valor is None or str(valor).strip() == "":
            ManejoErrores.advertencia(padre, "Campo requerido", f"El campo '{nombreCampo}' es obligatorio.")
            return False
        return True

    @staticmethod
    def validarNumero(padre, valor, nombreCampo):
        try:
            d = float(str(valor).strip().replace(",", "."))
            if d < 0:
                ManejoErrores.advertencia(padre, "Valor invalido", f"'{nombreCampo}' debe ser un numero positivo.")
                return -1.0
            return d
        except ValueError:
            ManejoErrores.advertencia(padre, "Formato incorrecto", f"'{nombreCampo}' debe ser un numero valido.")
            return -1.0

    @staticmethod
    def getRuta():
        return os.path.abspath(ManejoErrores.RUTA_LOG)

File: Util/ManejoErrores.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Microsoft.Maui.Controls;

namespace punto_de_venta_C_.Util
{
    public class ManejoErrores
    {
        private static readonly string RUTA_LOG = "corporativo_pos_errores.log";

        static ManejoErrores()
        {
        }

        public static async Task error(Page padre, string titulo, string mensaje)
        {
            registrar("SEVERE", titulo + ": " + mensaje);
            if (padre != null)
                await padre.DisplayAlert(titulo, mensaje, "OK");
        }

        public static async Task error(Page padre, string titulo, string mensaje, Exception ex)
        {
            registrarError(titulo + ": " + mensaje, ex);
            if (padre != null)
                await padre.DisplayAlert(titulo, mensaje + "\n\nDetalle: " + ex.Message, "OK");
        }

        public static async Task advertencia(Page padre, string titulo, string mensaje)
        {
            registrar("WARNING", titulo + ": " + mensaje);
            if (padre != null)
                await padre.DisplayAlert(titulo, mensaje, "OK");
        }

        public static async Task info(Page padre, string titulo, string mensaje)
        {
            if (padre != null)
                await padre.DisplayAlert(titulo, mensaje, "OK");
        }

        public static async Task<bool> confirmar(Page padre, string titulo, string mensaje)
        {
            if (padre != null)
                return await padre.DisplayAlert(titulo, mensaje, "Sí", "No");
            return false;
        }

        public static void registrar(string nivel, string mensaje)
        {
            try
            {
                string log = $"[{DateTime.Now:dd/MM/yyyy HH:mm:ss}] [{nivel}] {mensaje}\n";
                File.AppendAllText(RUTA_LOG, log);
            }
            catch { }
        }

        public static void registrarError(string contexto, Exception ex)
        {
            registrar("SEVERE", contexto + "\n" + ex.ToString());
        }

        public static void registrarInfo(string evento)
        {
            registrar("INFO", evento);
        }

        public static async Task<bool> validarRequerido(Page padre, string valor, string nombreCampo)
        {
            if (string.IsNullOrWhiteSpace(valor))
            {
                await advertencia(padre, "Campo requerido", "El campo '" + nombreCampo + "' es obligatorio.");
                return false;
            }
            return true;
        }

        public static async Task<double> validarNumero(Page padre, string valor, string nombreCampo)
        {
            try
            {
                double d = double.Parse(valor.Trim().Replace(",", "."));
                if (d < 0)
                {
                    await advertencia(padre, "Valor invalido", "'" + nombreCampo + "' debe ser un numero positivo.");
                    return -1;
                }
                return d;
            }
            catch (FormatException)
            {
                await advertencia(padre, "Formato incorrecto", "'" + nombreCampo + "' debe ser un numero valido.");
                return -1;
            }
        }

        public static string getRuta()
        {
            return Path.GetFullPath(RUTA_LOG);
        }
    }
}
